fix: Use single-file-artifact-gen in Create an Artifact (#1953)
mendersoftware/mender-docs: 06.Artifact-creation/01.Create-an-Artifact/docs.md

Creating an Artifact takes a different form in the case of [application updates](../../02.Overview/01.Introduction/docs.md#application-updates).

For application updates, the command for `mender-artifact` tool will be different depending on the

> implementation details of the actual update module. Most of the Update Modules come with an script

Reviewer: ```suggestion
implementation details of the actual update module. Most of the Update Modules come with a script
```

> to simplify the Artifact creation so that the implementation details are hidden to the final user.

Reviewer: ```suggestion
to simplify the Artifact creation so that the implementation details are hidden from the final user.
```

> For example, assume that you want copy a new `authorized_keys` file to the `/home/${USER}/.ssh`

Reviewer: ```suggestion
For example, assume that you want to copy a new `authorized_keys` file to the `/home/${USER}/.ssh`
```

directory on your devices, where `USER` holds your user name.

> For this example, we are creating a *module-image*, using the [single file](https://hub.mender.io/t/single-file/486/26?target=_blank) [Update Module](../../06.Artifact-creation/08.Create-a-custom-Update-Module/docs.md).

Reviewer 2: ```suggestion
We will use the [single file](https://hub.mender.io/t/single-file/486/26?target=_blank) [Update Module](../../06.Artifact-creation/08.Create-a-custom-Update-Module/docs.md) to create a *module-image*.
```

First, download the script to generate the Artifact and make it executable:
<!--AUTOVERSION: "mendersoftware/mender/%/support"/mender-->
```
curl -O https://raw.githubusercontent.com/mendersoftware/mender/master/support/modules-artifact-gen/single-file-artifact-gen
chmod +x single-file-artifact-gen
```

Now create the Artifact with:
```bash
./single-file-artifact-gen \
  --device-type raspberrypi4 \
  -o artifact.mender \
  -n updated-authorized_keys-1.0 \
  --software-name authorized_keys \
  --software-version 1.0 \
  --dest-dir /home/${USER}/.ssh \
  authorized_keys
```

Note specifically that in this case we are creating a *module-image*, using the [single
file](https://hub.mender.io/t/single-file/486/26?target=_blank) [Update
Module](../../06.Artifact-creation/08.Create-a-custom-Update-Module/docs.md). The Artifact created

> will be compatible with the *raspberrypi4* device type, although you can specify multiple device

Reviewer 2: can you clarify how? It is not clear if passing multiple `--device-type` flags or a comma separated list is the way to go.

Author: I'll try

types if needed. The name of the Artifact is declared as *updated-authorized_keys-1.0*, we set the
version of the software to *1.0* and indicate that it will be installed in the *rootfs* partition.
The resulting file `artifact.mender` holds the Artifact.

<!--AUTOVERSION: "mendersoftware/mender/blob/%/support"/mender-->

> You can also see how simple it is to [write a custom Update Module.](https://github.com/mendersoftware/mender/blob/master/support/modules/single-file?target=_blank)

Reviewer 2: ```suggestion
To get started writing your own custom one, please check out the [single file Update Module.](https://github.com/mendersoftware/mender/blob/master/support/modules/single-file?target=_blank)
```

Author: "To get started" I would point the users to [the tutorial in Mender Docs](https://docs.mender.io/artifact-creation/create-a-custom-update-module) rather than the source code of a "simple" update module.

I'll see how can I rephrase this to have both :)

Author: Okey, this is actually mentioned below.

I will them modify the text here to:

```suggestion
Inspect the source code of the [single file Update Module](https://github.com/mendersoftware/mender/blob/master/support/modules/single-file?target=_blank) to learn about implementation details of this module.
```
